ui/src/app/Pages/AddCandidateInfo/candidates/candidates.component.ts

```diff
@@ -0,0 +1,166 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { PageEvent } from '@angular/material/paginator';
+import { CandidateRepository } from 'src/app/domain/candidate/candidate.repository';
+import { Candidate } from 'src/app/domain/candidate/models/candidate';
+
+@Component({
+  selector: 'app-candidates',
+  templateUrl: './candidates.component.html',
+  styleUrls: ['./candidates.component.sass'],
+})
+export class CandidatesComponent implements OnInit {
+  candidateForm!: FormGroup;
+  candidateDetails!: Candidate;
+  fileAttr: string = 'Choose File';
+  pictureChanged: boolean = false;
+  dataimage!: any;
+  displayedColumns: string[] = [
+    'id',
+    'name',
+    'phone',
+    'email',
+    'delete',
+    'update',
+  ];
+  isVisible: boolean = false;
+  isAppear: boolean = false;
+
+  size: number = 10;
+  page: number = 0;
+  totalRows: number = 0;
+  allCandidates: Candidate[] = [];
+  submit: boolean = false;
+  submitted: boolean = false;
+  constructor(
+    private formBuilder: FormBuilder,
+    private candidateRepository: CandidateRepository
+  ) { }
+  ngOnInit(): void {
+    this.candidateDetails = { id: 0, name: '', phone: '', email: '', documentId: 0, picture: '' }
+
+    this.candidForm();
+    this.getAllCandidates();
+  }
+  candidForm() {
+    this.candidateForm = this.formBuilder.group({
+      id: [''],
+      name: ['', Validators.required],
+      phone: ['', Validators.required],
+      email: ['', Validators.required],
+      documentId: [''],
+      picture: [''],
+    });
+  }
+  saveCandidate() {
+    this.candidateDetails.id = this.candidateForm.value.id;
+    this.candidateDetails.name = this.candidateForm.value.name;
+    this.candidateDetails.phone = this.candidateForm.value.phone;
+    this.candidateDetails.email = this.candidateForm.value.email;
+    this.candidateDetails.documentId = this.candidateForm.value.documentId;
+    this.candidateDetails.picture = this.candidateForm.value.picture;
```

Comment:
you don't need to make this lines
you can send form.value direct

Reply:
I will do that.